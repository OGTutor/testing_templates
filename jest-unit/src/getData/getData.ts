import axios from 'axios';
import { mapArrToStrings } from '../mapArrToStrings/mapArrToStrings';

export type user = {
	userId: number;
	id: number;
	title: string;
	body: string;
};

export const getData = async () => {
	try {
		const response = await axios.get(
			'https://jsonplaceholder.typicode.com/posts'
		);
		const userIds = response.data.map((user: user) => user.id);
		return mapArrToStrings(userIds);
	} catch (error) {
		console.log(error);
	}
};
