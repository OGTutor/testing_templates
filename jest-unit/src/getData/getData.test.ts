import axios from 'axios';
import { getData, user } from './getData';

export type AxiosResponse = {
	data: user[];
};

jest.mock('axios');

describe('Get data', () => {
	let response: AxiosResponse;
	beforeEach(() => {
		response = {
			data: [
				{
					userId: 1,
					id: 1,
					title:
						'sunt aut facere repellat provident occaecati excepturi optio reprehenderit',
					body: 'quia et suscipit\nsuscipit recusandae consequuntur expedita et cum\nreprehenderit molestiae ut ut quas totam\nnostrum rerum est autem sunt rem eveniet architecto',
				},
				{
					userId: 1,
					id: 2,
					title: 'qui est esse',
					body: 'est rerum tempore vitae\nsequi sint nihil reprehenderit dolor beatae ea dolores neque\nfugiat blanditiis voluptate porro vel nihil molestiae ut reiciendis\nqui aperiam non debitis possimus qui neque nisi nulla',
				},
				{
					userId: 1,
					id: 3,
					title: 'ea molestias quasi exercitationem repellat qui ipsa sit aut',
					body: 'et iusto sed quo iure\nvoluptatem occaecati omnis eligendi aut ad\nvoluptatem doloribus vel accusantium quis pariatur\nmolestiae porro eius odio et labore et velit aut',
				},
				{
					userId: 1,
					id: 4,
					title: 'eum et est occaecati',
					body: 'ullam et saepe reiciendis voluptatem adipisci\nsit amet autem assumenda provident rerum culpa\nquis hic commodi nesciunt rem tenetur doloremque ipsam iure\nquis sunt voluptatem rerum illo velit',
				},
				{
					userId: 1,
					id: 5,
					title: 'nesciunt quas odio',
					body: 'repudiandae veniam quaerat sunt sed\nalias aut fugiat sit autem sed est\nvoluptatem omnis possimus esse voluptatibus quis\nest aut tenetur dolor neque',
				},
			],
		};
	});
	test('Correct value', async () => {
		(axios.get as jest.MockedFunction<typeof axios.get>).mockResolvedValue(
			response
		);
		const data = await getData();
		expect(axios.get).toHaveBeenCalledTimes(1);
		expect(data).toEqual(['1', '2', '3', '4', '5']);
		expect(data).toMatchSnapshot();
	});
});
