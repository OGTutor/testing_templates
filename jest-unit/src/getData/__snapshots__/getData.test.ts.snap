// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Get data Correct value 1`] = `
[
  "1",
  "2",
  "3",
  "4",
  "5",
]
`;
